### PR TITLE
Fixes proof pipelining to store event commitments

### PR DESCRIPTION
**Pull request by jorgeantonio21**
This PR allows to store event commitments to a DB, from a substreams block stream. 

How to demo: The code is in `tests/commitments/` and detailed instructions can be found in the folder `README.md`.

### CODE REVIEW

**Comment by jorgeantonio21** [comment]
Resolves #93

File: bin/node/cli/src/main.rs [hunk 1]
```diff
@@ -247,7 +261,9 @@ async fn main() {
                                 }
                             }
                         }
-                        _ => {}
+                        _ => {
+                            println!("FLAG:     HERREEEE");
```
Comment (benjaminbollen) [inline bin/node/cli/src/main.rs]: let s remove this; or put a proper debug log

File: bin/node/cli/src/main.rs [hunk 2]
```diff
@@ -258,11 +274,24 @@ async fn main() {
                 // todo: can we flush here per block?
                 // db_loader.flush()
             }
-            _ => {}
+            Message::DebugSnapshotComplete(_) => {
```
Comment (benjaminbollen) [inline bin/node/cli/src/main.rs]: same, please remove or use logger

File: bin/node/cli/src/main.rs [hunk 3]
```diff
@@ -228,10 +239,13 @@ async fn main() {
...
-                                                    .add_task(&config.schema, request)
-                                                    .await
-                                                    .expect("Failed to add task.");
+                                                match resolver {
+                                                    Some(ref mut r) => r
+                                                        .add_task(&config.schema, request)
+                                                        .await
+                                                        .expect("Failed to add task."),
+                                                    None => (),
+                                                }
```
Comment (vadim-frolov) [inline bin/node/cli/src/main.rs]: nit style sync: wouldn't it be more clear/concise to use `if let Some() {}` when we only need one branch?

File: tests/commitments/src/block_commit.rs [hunk 4]
```diff
@@ -126,12 +126,24 @@ impl BlockCommitment {
...
+            let diff = 2_u64.pow(log_2_len) - self.encoded_logs.len() as u64 - 1;
+
+            let mut extended_events_commitment = vec![vec![F::ZERO]; diff as usize];
+            // prepend with goldilocks_encoding length
+            extended_events_commitment
+                .push(vec![F::from_canonical_u64(self.encoded_logs.len() as u64)]);
+            let _ = self
+                .encoded_logs
+                .iter()
+                .for_each(|l| extended_events_commitment.push(l.goldilock_encoding.clone()));
```
Comment (benjaminbollen) [inline tests/commitments/src/block_commit.rs:144]: do we now not create a vector `[<0>, <0>, ..., <0>, len,  <ev1>, <ev2>, ...]`?

**Review by benjaminbollen — CHANGES_REQUESTED** [review]
please remove (or add logger) where indicated;

can you comment on the order of padding and events in the trie?

File: bin/node/cli/src/main.rs (hunk 3 above)
Comment (jorgeantonio21) [reply bin/node/cli/src/main.rs]: Absolutely ! Much preferred :)

File: bin/node/cli/src/main.rs (hunk 1 above)
Comment (jorgeantonio21) [reply bin/node/cli/src/main.rs]: Absolutely, left overs from yesterday's long debugging

File: tests/commitments/src/block_commit.rs (hunk 4 above)
Comment (jorgeantonio21) [reply tests/commitments/src/block_commit.rs:144]: This is precisely what we do, right ? We prepend the original event commitment leaves with `<0>`'s and `len` to generate a power of 2 leaf vector

File: bin/node/cli/src/main.rs (hunk 3 above)
Comment (benjaminbollen) [reply bin/node/cli/src/main.rs]: this is old-school me from 2014 writing `match` everywhere

File: tests/commitments/src/block_commit.rs (hunk 4 above)
Comment (benjaminbollen) [reply tests/commitments/src/block_commit.rs:144]: right, but postpending it is easier no?, maybe it doesnt matter in the end, but we'd always have to calculate this offset to find which path in the trie we need. Just feels a bit weird; maybe for now this works

**Review by benjaminbollen — APPROVED** [review]
LGTM

**Pull request merged**